lib/cloudregister/registerutils.py

```diff
@@ -1313,8 +1327,23 @@ def uses_rmt_as_scc_proxy():
 
 
 # ----------------------------------------------------------------------------
-def get_state_dir():
-    return REGISTRATION_DATA_DIR
+def write_framework_identifier(cfg):
+    """Write a file we use as identifier to detect movement of a golden
+       image created from a registered instance"""
+    identifier = {}
+    vendor, error = exec_subprocess(
```

Comment:
do we not check error value ?

Reply:
or that `vendor` is not -1  ?
which , in case `exec_subprocess` failed with `OSError`, that line would generate a `TypeError` 
because it can't unpack two values

Reply:
It is expected that the return code is **not** `-1`, i.e. everything works. The `TypeError` is triggered if `-1` is returned because `-1` is not iterable. That said, yes a total failure of the `dmidecode` command should be handled gracefully.